pkg/scheduler/schedulercache/cache_test.go

```diff
@@ -992,62 +1050,67 @@ func TestNodeOperators(t *testing.T) {
 		},
 	}
 
-	for _, test := range tests {
-		expected := buildNodeInfo(test.node, test.pods)
-		node := test.node
-
-		cache := newSchedulerCache(time.Second, time.Second, nil)
-		cache.AddNode(node)
-		for _, pod := range test.pods {
-			cache.AddPod(pod)
-		}
-
-		// Case 1: the node was added into cache successfully.
-		got, found := cache.nodes[node.Name]
-		if !found {
-			t.Errorf("Failed to find node %v in schedulercache.", node.Name)
-		}
-
-		// Generations are globally unique. We check in our unit tests that they are incremented correctly.
-		expected.generation = got.generation
-		if !reflect.DeepEqual(got, expected) {
-			t.Errorf("Failed to add node into schedulercache:\n got: %+v \nexpected: %+v", got, expected)
-		}
-
-		// Case 2: dump cached nodes successfully.
-		cachedNodes := map[string]*NodeInfo{}
-		cache.UpdateNodeNameToInfoMap(cachedNodes)
-		newNode, found := cachedNodes[node.Name]
-		if !found || len(cachedNodes) != 1 {
-			t.Errorf("failed to dump cached nodes:\n got: %v \nexpected: %v", cachedNodes, cache.nodes)
-		}
-		expected.generation = newNode.generation
-		if !reflect.DeepEqual(newNode, expected) {
-			t.Errorf("Failed to clone node:\n got: %+v, \n expected: %+v", newNode, expected)
-		}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("node/%v", i), func(t *testing.T) {
+			expected := buildNodeInfo(test.node, test.pods)
+			node := test.node
+			cache := newSchedulerCache(time.Second, time.Second, nil)
+			cache.AddNode(node)
+			for _, pod := range test.pods {
+				cache.AddPod(pod)
+			}
+			got, found := cache.nodes[node.Name]
+			t.Run("Case 1: the node was added into cache successfully.", func(t *testing.T) {
```

Comment:
I actually don't think this test benefits from sub-subtests. The three cases are sequential operations with distinct failure messages.

Reply:
my thinking here was that rather than have comments in the code, we could incorporate the context description from the comments into the test, using the subtest name.

that said, its a bit out of scope, since its not really in the spirit of sub tests for the individual tests in a table. Happy to remove it and leave such rework to other PRs and other discussions.

let me know.